src/searchbar/SearchBar-ios.js

```diff
@@ -155,11 +164,33 @@ class SearchBar extends Component {
             this.setState({ cancelButtonWidth: event.nativeEvent.layout.width })
           }
         >
-          <Button
-            title={cancelButtonTitle}
+          <TouchableOpacity
+            accessibilityRole="button"
             onPress={this.cancel}
-            {...cancelButtonProps}
-          />
+            disabled={buttonDisabled}
+            {...otherCancelButtonProps}
+          >
+            <View
+              style={[
+                buttonColor &&
+                  Platform.OS === 'android' && { backgroundColor: buttonColor },
```

Comment:
Actually as we said you shouldn't have any differences between iOS and Android here (no background on iOS, neither on Android)